tests/schema/tests.py

```diff
@@ -2303,6 +2304,54 @@ class Meta:
         columns = self.column_classes(Author)
         self.assertEqual(columns["birth_year"][1].default, "1988")
 
+    @isolate_apps("schema")
+    def test_add_text_field_with_db_default(self):
+        class Author(Model):
+            description = TextField(db_default="(missing)")
+
+            class Meta:
+                app_label = "schema"
+
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+
+    @isolate_apps("schema")
+    def test_db_default_equivalent_sql_noop(self):
+        class Author(Model):
+            name = TextField(db_default=Value("foo"))
+
+            class Meta:
+                app_label = "schema"
+
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+
+        new_field = TextField(db_default="foo")
+        new_field.set_attributes_from_name("name")
+        new_field.model = Author
+        with connection.schema_editor() as editor, self.assertNumQueries(0):
+            editor.alter_field(Author, Author._meta.get_field("name"), new_field)
+
+    @isolate_apps("schema")
+    def test_db_default_output_field_resolving(self):
+        class Author(Model):
+            data = JSONField(
+                encoder=DjangoJSONEncoder,
+                db_default={
+                    "epoch": datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc)
+                },
+            )
+
+            class Meta:
+                app_label = "schema"
+
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+
+        author = Author.objects.create()
+        author.refresh_from_db()
```

Comment:
I was surprised that on backends that don't support the `RETURNING` clause we don't defer the fields instead of keeping `DatabaseDefault` instances assigned.

I guess that's a larger problem than with `db_default` though as the same happens when doing attribute assignments of expressions on a model instance and calling `save` on it.